core/core-impl/src/main/java/io/openshift/appdev/missioncontrol/core/impl/MissionControlImpl.java

```diff
@@ -186,23 +211,20 @@ private String getUserId(Projectile projectile) {
         return userId;
     }
 
-    private GitHubWebhook getGitHubWebhook(GitHubService gitHubService, OpenShiftService openShiftService,
+    private List<GitHubWebhook> getGitHubWebhooks(GitHubService gitHubService, OpenShiftService openShiftService,
                                            GitHubRepository gitHubRepository, OpenShiftProject createdProject) {
-        GitHubWebhook webhook;
-        final URL webhookUrl = openShiftService.getWebhookUrl(createdProject);
-        if (webhookUrl != null) {
-            try {
-                webhook = gitHubService.createWebhook(gitHubRepository, webhookUrl, GitHubWebhookEvent.PUSH);
-            } catch (final DuplicateWebhookException dpe) {
-                // Swallow, it's OK, we've already forked this repo
-                log.log(Level.INFO, dpe.getMessage());
-                webhook = ((GitHubServiceSpi) gitHubService)
-                        .getWebhook(gitHubRepository, webhookUrl);
-            }
-        } else {
-            webhook = null;
-        }
-        return webhook;
+        List<GitHubWebhook> webhooks = openShiftService.getWebhookUrls(createdProject).stream()
+                .map(webhookUrl -> {
+                    try {
+                        return gitHubService.createWebhook(gitHubRepository, webhookUrl, GitHubWebhookEvent.PUSH);
+                    } catch (final DuplicateWebhookException dpe) {
```

Comment:
I don't understand, why create a new webhook for each existing webhook in the github repository?